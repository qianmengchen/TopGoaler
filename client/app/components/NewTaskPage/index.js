import NewTaskPage from './container';

export { NewTaskPage };
