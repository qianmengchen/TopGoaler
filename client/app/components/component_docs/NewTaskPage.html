<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: NewTaskPage</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: NewTaskPage</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>NewTaskPage<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="NewTaskPage"><span class="type-signature"></span>new NewTaskPage<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>user_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">A unique number identifying a user.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>channel_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last">A unique number identifying a channel.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>addProposal</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last">A function allowing the user to create a proposal for a new task within a channel.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="NewTaskPage.js.html">NewTaskPage.js</a>, <a href="NewTaskPage.js.html#line51">line 51</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="NewTaskPage.html">NewTaskPage</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sun Dec 02 2018 21:03:40 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>