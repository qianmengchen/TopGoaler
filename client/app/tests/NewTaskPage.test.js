import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import { Input, Button } from 'react-native-elements';
import RNPickerSelect from 'react-native-picker-select';
import sinon from 'sinon';
require('isomorphic-fetch');

import NewTaskPage from '../components/NewTaskPage/NewTaskPage';
const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing New Task Page', () => {
  const updateName = sinon.spy(NewTaskPage.prototype, '_updateName');
  const updateDescription = sinon.spy(
    NewTaskPage.prototype,
    '_updateDescription'
  );
  const updatePeriod = sinon.spy(NewTaskPage.prototype, '_updatePeriod');
  const updatePattern = sinon.spy(NewTaskPage.prototype, '_updatePattern');
  const submit = sinon.spy(NewTaskPage.prototype, '_submit');
  const navigation = { navigate: jest.fn() };

  const wrapper = shallow(<NewTaskPage navigation={navigation} />, {
    context: { store: mockStore(initialTaskState) }
  });
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should correctly update name and description states on text change', () => {
    render.find(Input).forEach(child => {
      child.simulate('changeText', 'new text');
    });

    expect(updateName.calledOnce).toBe(true);
    expect(updateDescription.calledOnce).toBe(true);
  });

  it('should correctly update period and pattern states on text change', () => {
    wrapper.find(RNPickerSelect).forEach(child => {
      child.simulate('valueChange', 'new value');
    });

    expect(updatePeriod.calledOnce).toBe(true);
    expect(updatePattern.calledOnce).toBe(true);
  });

  it('should invoke correct methods when pressing submit button', () => {
    wrapper.find(Button).forEach(child => {
      child.simulate('press');
    });

    expect(submit.calledOnce).toBe(true);
  });
});
