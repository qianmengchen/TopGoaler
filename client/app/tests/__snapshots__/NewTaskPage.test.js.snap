// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing New Task Page renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginVertical": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        New Task
      </Text>
    </Component>
    <ForwardRef(Themed.Input)
      containerStyle={
        Object {
          "marginBottom": 10,
          "paddingLeft": 10,
        }
      }
      errorMessage={null}
      errorStyle={
        Object {
          "color": "red",
        }
      }
      label="Name"
      labelStyle={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "300",
        }
      }
      onChangeText={[Function]}
      value=""
    />
    <ForwardRef(Themed.Input)
      containerStyle={
        Object {
          "marginBottom": 10,
          "paddingLeft": 10,
        }
      }
      errorMessage={null}
      errorStyle={
        Object {
          "color": "red",
        }
      }
      label="Description"
      labelStyle={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "300",
        }
      }
      onChangeText={[Function]}
      value=""
    />
    <RNPickerSelect
      disabled={false}
      doneText="Done"
      hideDoneBar={false}
      hideIcon={false}
      itemKey={null}
      items={
        Array [
          Object {
            "label": "Daily",
            "value": "Daily",
          },
          Object {
            "label": "Weekly",
            "value": "Weekly",
          },
          Object {
            "label": "Monthly",
            "value": "Monthly",
          },
          Object {
            "label": "One Time Only",
            "value": "One Time Only",
          },
        ]
      }
      modalProps={Object {}}
      onDonePress={null}
      onDownArrow={null}
      onUpArrow={null}
      onValueChange={[Function]}
      pickerProps={Object {}}
      placeholder={
        Object {
          "label": "Select something",
          "value": null,
        }
      }
      placeholderTextColor="#C7C7CD"
      style={Object {}}
      textInputProps={Object {}}
      useNativeAndroidPickerStyle={true}
      value=""
    >
      <ForwardRef(Themed.Input)
        containerStyle={
          Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          }
        }
        label="Period"
        labelStyle={
          Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          }
        }
        placeholder="Select a period..."
        placeholderTextColor="gray"
        value=""
      />
    </RNPickerSelect>
    <RNPickerSelect
      disabled={false}
      doneText="Done"
      hideDoneBar={false}
      hideIcon={false}
      itemKey={null}
      items={
        Array [
          Object {
            "label": "Once",
            "value": "Once",
          },
          Object {
            "label": "Twice",
            "value": "Twice",
          },
          Object {
            "label": "Three times",
            "value": "Three times",
          },
          Object {
            "label": "Unlimited",
            "value": "Unlimited",
          },
        ]
      }
      modalProps={Object {}}
      onDonePress={null}
      onDownArrow={null}
      onUpArrow={null}
      onValueChange={[Function]}
      pickerProps={Object {}}
      placeholder={Object {}}
      placeholderTextColor="#C7C7CD"
      style={Object {}}
      textInputProps={Object {}}
      useNativeAndroidPickerStyle={true}
    >
      <ForwardRef(Themed.Input)
        containerStyle={
          Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          }
        }
        label="Pattern"
        labelStyle={
          Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          }
        }
        placeholder="Select a pattern..."
        placeholderTextColor="gray"
        value=""
      />
    </RNPickerSelect>
    <Component
      style={
        Object {
          "alignItems": "center",
          "marginVertical": 10,
        }
      }
    >
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "rgba(40, 160, 147, 1)",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderWidth": 0,
            "height": 40,
            "width": 120,
          }
        }
        onPress={[Function]}
        title="Create"
      />
    </Component>
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
              }
            }
          >
            New Task
          </Text>
        </Component>,
        <ForwardRef(Themed.Input)
          containerStyle={
            Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            }
          }
          errorMessage={null}
          errorStyle={
            Object {
              "color": "red",
            }
          }
          label="Name"
          labelStyle={
            Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            }
          }
          onChangeText={[Function]}
          value=""
        />,
        <ForwardRef(Themed.Input)
          containerStyle={
            Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            }
          }
          errorMessage={null}
          errorStyle={
            Object {
              "color": "red",
            }
          }
          label="Description"
          labelStyle={
            Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            }
          }
          onChangeText={[Function]}
          value=""
        />,
        <RNPickerSelect
          disabled={false}
          doneText="Done"
          hideDoneBar={false}
          hideIcon={false}
          itemKey={null}
          items={
            Array [
              Object {
                "label": "Daily",
                "value": "Daily",
              },
              Object {
                "label": "Weekly",
                "value": "Weekly",
              },
              Object {
                "label": "Monthly",
                "value": "Monthly",
              },
              Object {
                "label": "One Time Only",
                "value": "One Time Only",
              },
            ]
          }
          modalProps={Object {}}
          onDonePress={null}
          onDownArrow={null}
          onUpArrow={null}
          onValueChange={[Function]}
          pickerProps={Object {}}
          placeholder={
            Object {
              "label": "Select something",
              "value": null,
            }
          }
          placeholderTextColor="#C7C7CD"
          style={Object {}}
          textInputProps={Object {}}
          useNativeAndroidPickerStyle={true}
          value=""
        >
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            label="Period"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            placeholder="Select a period..."
            placeholderTextColor="gray"
            value=""
          />
        </RNPickerSelect>,
        <RNPickerSelect
          disabled={false}
          doneText="Done"
          hideDoneBar={false}
          hideIcon={false}
          itemKey={null}
          items={
            Array [
              Object {
                "label": "Once",
                "value": "Once",
              },
              Object {
                "label": "Twice",
                "value": "Twice",
              },
              Object {
                "label": "Three times",
                "value": "Three times",
              },
              Object {
                "label": "Unlimited",
                "value": "Unlimited",
              },
            ]
          }
          modalProps={Object {}}
          onDonePress={null}
          onDownArrow={null}
          onUpArrow={null}
          onValueChange={[Function]}
          pickerProps={Object {}}
          placeholder={Object {}}
          placeholderTextColor="#C7C7CD"
          style={Object {}}
          textInputProps={Object {}}
          useNativeAndroidPickerStyle={true}
        >
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            label="Pattern"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            placeholder="Select a pattern..."
            placeholderTextColor="gray"
            value=""
          />
        </RNPickerSelect>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "marginVertical": 10,
            }
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              }
            }
            onPress={[Function]}
            title="Create"
          />
        </Component>,
      ],
      "style": Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
              }
            }
          >
            New Task
          </Text>,
          "style": Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 20,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "New Task",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": "New Task",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          "errorMessage": null,
          "errorStyle": Object {
            "color": "red",
          },
          "label": "Name",
          "labelStyle": Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          },
          "onChangeText": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          "errorMessage": null,
          "errorStyle": Object {
            "color": "red",
          },
          "label": "Description",
          "labelStyle": Object {
            "color": "grey",
            "fontSize": 15,
            "fontWeight": "300",
          },
          "onChangeText": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            label="Period"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            placeholder="Select a period..."
            placeholderTextColor="gray"
            value=""
          />,
          "disabled": false,
          "doneText": "Done",
          "hideDoneBar": false,
          "hideIcon": false,
          "itemKey": null,
          "items": Array [
            Object {
              "label": "Daily",
              "value": "Daily",
            },
            Object {
              "label": "Weekly",
              "value": "Weekly",
            },
            Object {
              "label": "Monthly",
              "value": "Monthly",
            },
            Object {
              "label": "One Time Only",
              "value": "One Time Only",
            },
          ],
          "modalProps": Object {},
          "onDonePress": null,
          "onDownArrow": null,
          "onUpArrow": null,
          "onValueChange": [Function],
          "pickerProps": Object {},
          "placeholder": Object {
            "label": "Select something",
            "value": null,
          },
          "placeholderTextColor": "#C7C7CD",
          "style": Object {},
          "textInputProps": Object {},
          "useNativeAndroidPickerStyle": true,
          "value": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "label": "Period",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "placeholder": "Select a period...",
            "placeholderTextColor": "gray",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            label="Pattern"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            placeholder="Select a pattern..."
            placeholderTextColor="gray"
            value=""
          />,
          "disabled": false,
          "doneText": "Done",
          "hideDoneBar": false,
          "hideIcon": false,
          "itemKey": null,
          "items": Array [
            Object {
              "label": "Once",
              "value": "Once",
            },
            Object {
              "label": "Twice",
              "value": "Twice",
            },
            Object {
              "label": "Three times",
              "value": "Three times",
            },
            Object {
              "label": "Unlimited",
              "value": "Unlimited",
            },
          ],
          "modalProps": Object {},
          "onDonePress": null,
          "onDownArrow": null,
          "onUpArrow": null,
          "onValueChange": [Function],
          "pickerProps": Object {},
          "placeholder": Object {},
          "placeholderTextColor": "#C7C7CD",
          "style": Object {},
          "textInputProps": Object {},
          "useNativeAndroidPickerStyle": true,
          "value": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "label": "Pattern",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "placeholder": "Select a pattern...",
            "placeholderTextColor": "gray",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              }
            }
            onPress={[Function]}
            title="Create"
          />,
          "style": Object {
            "alignItems": "center",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "rgba(40, 160, 147, 1)",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderWidth": 0,
              "height": 40,
              "width": 120,
            },
            "onPress": [Function],
            "title": "Create",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginVertical": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 30,
                  "fontWeight": "bold",
                }
              }
            >
              New Task
            </Text>
          </Component>,
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            errorMessage={null}
            errorStyle={
              Object {
                "color": "red",
              }
            }
            label="Name"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            onChangeText={[Function]}
            value=""
          />,
          <ForwardRef(Themed.Input)
            containerStyle={
              Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              }
            }
            errorMessage={null}
            errorStyle={
              Object {
                "color": "red",
              }
            }
            label="Description"
            labelStyle={
              Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              }
            }
            onChangeText={[Function]}
            value=""
          />,
          <RNPickerSelect
            disabled={false}
            doneText="Done"
            hideDoneBar={false}
            hideIcon={false}
            itemKey={null}
            items={
              Array [
                Object {
                  "label": "Daily",
                  "value": "Daily",
                },
                Object {
                  "label": "Weekly",
                  "value": "Weekly",
                },
                Object {
                  "label": "Monthly",
                  "value": "Monthly",
                },
                Object {
                  "label": "One Time Only",
                  "value": "One Time Only",
                },
              ]
            }
            modalProps={Object {}}
            onDonePress={null}
            onDownArrow={null}
            onUpArrow={null}
            onValueChange={[Function]}
            pickerProps={Object {}}
            placeholder={
              Object {
                "label": "Select something",
                "value": null,
              }
            }
            placeholderTextColor="#C7C7CD"
            style={Object {}}
            textInputProps={Object {}}
            useNativeAndroidPickerStyle={true}
            value=""
          >
            <ForwardRef(Themed.Input)
              containerStyle={
                Object {
                  "marginBottom": 10,
                  "paddingLeft": 10,
                }
              }
              label="Period"
              labelStyle={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
              placeholder="Select a period..."
              placeholderTextColor="gray"
              value=""
            />
          </RNPickerSelect>,
          <RNPickerSelect
            disabled={false}
            doneText="Done"
            hideDoneBar={false}
            hideIcon={false}
            itemKey={null}
            items={
              Array [
                Object {
                  "label": "Once",
                  "value": "Once",
                },
                Object {
                  "label": "Twice",
                  "value": "Twice",
                },
                Object {
                  "label": "Three times",
                  "value": "Three times",
                },
                Object {
                  "label": "Unlimited",
                  "value": "Unlimited",
                },
              ]
            }
            modalProps={Object {}}
            onDonePress={null}
            onDownArrow={null}
            onUpArrow={null}
            onValueChange={[Function]}
            pickerProps={Object {}}
            placeholder={Object {}}
            placeholderTextColor="#C7C7CD"
            style={Object {}}
            textInputProps={Object {}}
            useNativeAndroidPickerStyle={true}
          >
            <ForwardRef(Themed.Input)
              containerStyle={
                Object {
                  "marginBottom": 10,
                  "paddingLeft": 10,
                }
              }
              label="Pattern"
              labelStyle={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
              placeholder="Select a pattern..."
              placeholderTextColor="gray"
              value=""
            />
          </RNPickerSelect>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "marginVertical": 10,
              }
            }
          >
            <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "rgba(40, 160, 147, 1)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "height": 40,
                  "width": 120,
                }
              }
              onPress={[Function]}
              title="Create"
            />
          </Component>,
        ],
        "style": Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 30,
                  "fontWeight": "bold",
                }
              }
            >
              New Task
            </Text>,
            "style": Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 20,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "New Task",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 30,
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": "New Task",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "errorMessage": null,
            "errorStyle": Object {
              "color": "red",
            },
            "label": "Name",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "onChangeText": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "marginBottom": 10,
              "paddingLeft": 10,
            },
            "errorMessage": null,
            "errorStyle": Object {
              "color": "red",
            },
            "label": "Description",
            "labelStyle": Object {
              "color": "grey",
              "fontSize": 15,
              "fontWeight": "300",
            },
            "onChangeText": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ForwardRef(Themed.Input)
              containerStyle={
                Object {
                  "marginBottom": 10,
                  "paddingLeft": 10,
                }
              }
              label="Period"
              labelStyle={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
              placeholder="Select a period..."
              placeholderTextColor="gray"
              value=""
            />,
            "disabled": false,
            "doneText": "Done",
            "hideDoneBar": false,
            "hideIcon": false,
            "itemKey": null,
            "items": Array [
              Object {
                "label": "Daily",
                "value": "Daily",
              },
              Object {
                "label": "Weekly",
                "value": "Weekly",
              },
              Object {
                "label": "Monthly",
                "value": "Monthly",
              },
              Object {
                "label": "One Time Only",
                "value": "One Time Only",
              },
            ],
            "modalProps": Object {},
            "onDonePress": null,
            "onDownArrow": null,
            "onUpArrow": null,
            "onValueChange": [Function],
            "pickerProps": Object {},
            "placeholder": Object {
              "label": "Select something",
              "value": null,
            },
            "placeholderTextColor": "#C7C7CD",
            "style": Object {},
            "textInputProps": Object {},
            "useNativeAndroidPickerStyle": true,
            "value": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "containerStyle": Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              },
              "label": "Period",
              "labelStyle": Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              },
              "placeholder": "Select a period...",
              "placeholderTextColor": "gray",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ForwardRef(Themed.Input)
              containerStyle={
                Object {
                  "marginBottom": 10,
                  "paddingLeft": 10,
                }
              }
              label="Pattern"
              labelStyle={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                  "fontWeight": "300",
                }
              }
              placeholder="Select a pattern..."
              placeholderTextColor="gray"
              value=""
            />,
            "disabled": false,
            "doneText": "Done",
            "hideDoneBar": false,
            "hideIcon": false,
            "itemKey": null,
            "items": Array [
              Object {
                "label": "Once",
                "value": "Once",
              },
              Object {
                "label": "Twice",
                "value": "Twice",
              },
              Object {
                "label": "Three times",
                "value": "Three times",
              },
              Object {
                "label": "Unlimited",
                "value": "Unlimited",
              },
            ],
            "modalProps": Object {},
            "onDonePress": null,
            "onDownArrow": null,
            "onUpArrow": null,
            "onValueChange": [Function],
            "pickerProps": Object {},
            "placeholder": Object {},
            "placeholderTextColor": "#C7C7CD",
            "style": Object {},
            "textInputProps": Object {},
            "useNativeAndroidPickerStyle": true,
            "value": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "containerStyle": Object {
                "marginBottom": 10,
                "paddingLeft": 10,
              },
              "label": "Pattern",
              "labelStyle": Object {
                "color": "grey",
                "fontSize": 15,
                "fontWeight": "300",
              },
              "placeholder": "Select a pattern...",
              "placeholderTextColor": "gray",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ForwardRef(Themed.Button)
              buttonStyle={
                Object {
                  "backgroundColor": "rgba(40, 160, 147, 1)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "height": 40,
                  "width": 120,
                }
              }
              onPress={[Function]}
              title="Create"
            />,
            "style": Object {
              "alignItems": "center",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonStyle": Object {
                "backgroundColor": "rgba(40, 160, 147, 1)",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 0,
                "height": 40,
                "width": 120,
              },
              "onPress": [Function],
              "title": "Create",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
